bootstrap/mod/fabric/src/main/java/org/geysermc/geyser/platform/fabric/GeyserFabricBootstrap.java

```diff
@@ -73,22 +87,96 @@ public void onInitialize() {
+//                        }
+//
+//                        List<String> serverFields = List.of("server", "field_13995");
+//
+//                        Field serverField = null;
+//
+//                        for (String methodName : serverFields) {
+//                            try {
+//                                serverField = player.getClass().getDeclaredField(methodName);
+//                                serverField.setAccessible(true);
+//                                break;
+//                            } catch (NoSuchFieldException ignored) {}
+//                        }
+//
+//                        if (serverField == null) {
+//                            throw new RuntimeException("Unable to get server from ServerPlayer.");
+//                        }
+//
+//                        // Double casting as in older versions, player is instance of CommandSource, but not in modern versions
+//                        //noinspection RedundantCast
+//                        return new CommandSourceStack((CommandSource) (Object) player, (Vec3) positionField.get(player), player.getRotationVector(), (ServerLevel) levelMethod.invoke(player), player.getPermissionLevel(), player.getName().getString(), player.getDisplayName(), (MinecraftServer) serverField.get(player), player);
+//                    } catch (IllegalAccessException | InvocationTargetException e) {
+//                        throw new RuntimeException(e);
+//                    }
+//                }
+            };
+
+            var sourceConverter = CommandSourceConverter.layered(
                 CommandSourceStack.class,
                 id -> getServer().getPlayerList().getPlayer(id),
-                ServerPlayer::createCommandSourceStack,
+                stackCreator,
                 () -> getServer().createCommandSourceStack(), // NPE if method reference is used, since server is not available yet
                 ModCommandSource::new
-        );
-        CommandManager<GeyserCommandSource> cloud = new FabricServerCommandManager<>(
+            );
+            CommandManager<GeyserCommandSource> cloud = new FabricServerCommandManager<>(
                 ExecutionCoordinator.simpleCoordinator(),
                 sourceConverter
-        );
-        this.setCommandRegistry(new CommandRegistry(GeyserImpl.getInstance(), cloud, false)); // applying root permission would be a breaking change because we can't register permission defaults
+            );
+
+            this.setCommandRegistry(new CommandRegistry(GeyserImpl.getInstance(), cloud, false)); // applying root permission would be a breaking change because we can't register permission defaults
+        } else { // Fallback to a standalone command manager
+            CommandManager<GeyserCommandSource> cloud = new StandaloneCloudCommandManager(GeyserImpl.getInstance());
+
+            this.setCommandRegistry(new CommandRegistry(GeyserImpl.getInstance(), cloud));
+        }
```

Comment:
hmmm would really prefer to only fall back if there's no cloud 2.0 available for that version